vignettes/zip_zap_models.Rmd

```diff
@@ -0,0 +1,185 @@
+---
+title: "Devel: ZIP and ZAP models"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Devel: ZIP and ZAP models}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+header-includes:
+  - \newcommand{\bm}[1]{\boldsymbol{#1}}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  eval = FALSE
+)
+```
+
+(Vignette under construction!)
+
+```{r setup, eval=TRUE, echo=TRUE, include=TRUE}
+library(tidyverse)
+library(inlabru)
+library(terra)
+library(sf)
+library(RColorBrewer)
+```
+
+# Count model
+
+In addition to point process models, `inlabru` can handle models with positive integer response, e.g. abundance models where species counts are recorded in every observed location. Count models can be seen as coarse aggregations of point process models.
+
+The following example uses `gorillas` dataset. In order to obtain the counts data we rasterize the species count to the same granularity as the spatial covariates available for the `gorilla` data, and then aggregate the pixels to cover 25 times bigger area (5x5 pixels in original covariate raster dimensions). Finally, we mask the regions outside of study area.
+
+```{r fig-count-raster, fig.cap="Counts of gorilla nests", fig.width=7, fig.height=5,out.width="80%", fig.align='center'}
+data(gorillas_sf)
+nests <- gorillas_sf$nests
+mesh <- gorillas_sf$mesh
+boundary <- gorillas_sf$boundary
+gcov <- gorillas_sf_gcov()
+counts_rstr <- terra::rasterize(vect(nests), gcov, fun=sum, background=0) |> 
+  terra::aggregate(fact=4, fun=sum) |> mask(vect(boundary))
+plot(counts_rstr)
+counts_rstr <- counts_rstr |> cellSize(unit="km") |> c(counts_rstr)
```

Comment:
In orer to use the new (since R 4.1) pipe operator we'd need to bump the R version requirement to `>= 4.1` from the current `>= 3.6`, and that would prevent people installing on some slow-to-update HPC system etc, so I'm not ready to do that yet.  Please change to `%>%` and add `import(magrittr)` at the setup chunk.